lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -2258,18 +2258,7 @@ def scatter(self, xs, ys, zs=0, zdir='z', s=20, c=None, depthshade=True,
 
         had_data = self.has_data()
 
-        xs = np.ma.ravel(xs)
-        ys = np.ma.ravel(ys)
-        zs = np.ma.ravel(zs)
-        if xs.size != ys.size:
-            raise ValueError("Arguments 'xs' and 'ys' must be of same size.")
-        if xs.size != zs.size:
-            if zs.size == 1:
-                zs = np.tile(zs[0], xs.size)
-            else:
-                raise ValueError(("Argument 'zs' must be of same size as 'xs' "
-                    "and 'ys' or of size 1."))
-
+        xs, ys, zs = np.broadcast_arrays(*map(np.ma.ravel, [xs, ys, zs]))
```

Comment:
`broadcast_to` explicitly prints the nonmatching shape in the error message, but not `broadcast_arrays`.  I'll open an issue on numpy but I don't think this should hold up this PR.

(Note how just below we gain checking on the size of dx/dy/dz, which was not present before.)